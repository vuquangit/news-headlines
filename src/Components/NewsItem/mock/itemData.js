export const mockData = {
  source: {
    id: "the-irish-times",
    name: "The Irish Times",
  },
  author: "The Irish Times",
  title:
    "LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says",
  description:
    "70,000 applications made to Ireland on Call healthcare worker scheme",
  url:
    "https://www.irishtimes.com/news/ireland/irish-news/coronavirus-4-500-tests-to-be-carried-out-per-day-from-this-week-hse-says-1.4221258",
  urlToImage:
    "https://www.irishtimes.com/polopoly_fs/1.4221277.1586082453!/image/image.jpg",
  publishedAt: "2020-04-05T10:53:21.0103778Z",
  content:
    "A medic wearing personal protective equipment in Bergamo, Italy. Some of the consignment of PPE to Ireland last weekend was found by the HSE to be not suitable for use. Photograph: Getty",
};
