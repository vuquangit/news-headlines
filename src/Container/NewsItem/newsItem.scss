.news-item {
    $maxWidthScreen: 555px;

    display: flex;
    flex-direction: row;
    align-items: flex-start;
    justify-content: center;

    padding: 25px 0;
    border-bottom: 1px solid #ebebeb;

    @media (max-width:$maxWidthScreen) {
        flex-direction: column;
    }

    &__img {
        flex-grow: 0;

        @media (max-width:$maxWidthScreen) {
            width: 100%;
            // flex-grow: 1;
        }

        &--content {
            display: block;
            overflow: hidden;
            position: relative;
            background-size: cover;
            background-position: center top;
            background-repeat: no-repeat;

            width: 250px;
            padding-top: 65.52%;
            cursor: pointer;

            @media (max-width:$maxWidthScreen) {
                width: 100%;
                height: 250px;
                padding: 0;
            }
        }

        &--empty {
            position: absolute;
            top: 0;
            left: 0;
            display: flex;
            width: 100%;
            height: 100%;
            align-items: center;
            justify-content: center;
            background-image: linear-gradient(90deg, #ff0000 0, #cb16c6);

            span {
                font-size: 24px;
            }
        }
    }

    &__info {
        display: flex;
        flex-direction: column;
        flex-grow: 1;

        text-align: left;
        padding-left: 16px;

        @media (max-width:$maxWidthScreen) {
            padding-left: 0;
        }

        &--title {
            font-size: 20px;
            line-height: 26px;
            color: #333;
            text-decoration: none;
            display: block;
            margin-bottom: 10px;
            margin-top: -5px;
            cursor: pointer;

            @media (max-width:$maxWidthScreen) {
                margin-top: 10px;
            }
        }

        &--published-at {
            font-size: 14px;
            font-weight: 400;
            color: #777;
            margin-bottom: 13px;
        }

        &--description {
            font-size: 13px;
            // font-family: Arial;
            color: #4d4d4d;
            line-height: 18px;
            display: block;
        }
    }
}