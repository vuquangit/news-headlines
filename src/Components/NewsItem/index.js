import React, { useState, useCallback } from "react";
import moment from "moment";
import { Modal, Button } from "antd";

import NewsDetail from "Components/NewsDetail";
import "./newsItem.scss";

const NewsItem = (props) => {
  const {
    title = "",
    description = "",
    urlToImage = "",
    publishedAt = "",
    source = {},
  } = props;

  const [visibleModal, setVisibleModal] = useState(false);
  const handleShowModal = useCallback(() => {
    setVisibleModal(true);
  }, []);
  const handleCancelModal = useCallback(() => {
    setVisibleModal(false);
  }, []);

  return (
    <div className="news-item">
      <div className="news-item__img">
        {urlToImage ? (
          <div
            style={{ backgroundImage: `url(${urlToImage})` }}
            title={title}
            onClick={handleShowModal}
            className="news-item__img--content"
          />
        ) : (
          <div className="news-item__img--content" onClick={handleShowModal}>
            <div className="news-item__img--empty">
              <span>News Headlines</span>
            </div>
          </div>
        )}
      </div>
      <div className="news-item__info">
        <h3 onClick={handleShowModal} className="news-item__info--title">
          {title}
        </h3>
        <div>
          {source.name && (
            <span className="news-item__info--source">{source.name}</span>
          )}
          {publishedAt && (
            <span className="news-item__info--published-at">
              {moment(publishedAt).fromNow()}
            </span>
          )}
        </div>
        {description && (
          <span className="news-item__info--description">{description}</span>
        )}
      </div>
      <Modal
        title={null}
        visible={visibleModal}
        onCancel={handleCancelModal}
        className="modal-detail"
        footer={[
          <Button key="back" onClick={handleCancelModal}>
            Cancel
          </Button>,
        ]}
        centered
      >
        <NewsDetail {...props} />
      </Modal>
    </div>
  );
};

export default NewsItem;
