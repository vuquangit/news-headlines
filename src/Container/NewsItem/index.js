import React from "react";
import moment from "moment";

import "./newsItem.scss";

const NewsItem = ({
  author = "",
  title = "",
  description = "",
  url = "",
  urlToImage = "",
  publishedAt = "",
  content = ""
}) => {
  return (
    <div className="news-item">
      <div className="news-item__img">
        {urlToImage ? (
          <div
            style={{ backgroundImage: `url(${urlToImage})` }}
            title={title}
            className="news-item__img--content"
          />
        ) : (
          <div className="news-item__img--content">
            <div className="news-item__img--empty">
              <span>News Headlines</span>
            </div>
          </div>
        )}
      </div>
      <div className="news-item__info">
        <h3 className="news-item__info--title">{title}</h3>
        {publishedAt && (
          <span className="news-item__info--published-at">
            {moment(publishedAt).fromNow()}
          </span>
        )}
        {description && (
          <span className="news-item__info--description">{description}</span>
        )}
      </div>
    </div>
  );
};

export default NewsItem;
