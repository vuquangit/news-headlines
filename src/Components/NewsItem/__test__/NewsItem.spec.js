import React from "react";
import { render } from "@testing-library/react";

import NewsItem from "..";
import { mockData } from "../mock/itemData";

describe("News Item", () => {
  it("Test get default NewsItem no params ", () => {
    // test snapshot
    const wrapper = render(<NewsItem />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Test get default NewsItem with data ", () => {
    // test snapshot
    const wrapper = render(<NewsItem {...mockData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
