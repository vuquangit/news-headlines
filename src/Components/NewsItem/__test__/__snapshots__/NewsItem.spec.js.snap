// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Item Test get default NewsItem no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="news-item"
      >
        <div
          class="news-item__img"
        >
          <div
            class="news-item__img--content"
          >
            <div
              class="news-item__img--empty"
            >
              <span>
                News Headlines
              </span>
            </div>
          </div>
        </div>
        <div
          class="news-item__info"
        >
          <h3
            class="news-item__info--title"
          />
          <div>
            
          </div>
          
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="news-item"
    >
      <div
        class="news-item__img"
      >
        <div
          class="news-item__img--content"
        >
          <div
            class="news-item__img--empty"
          >
            <span>
              News Headlines
            </span>
          </div>
        </div>
      </div>
      <div
        class="news-item__info"
      >
        <h3
          class="news-item__info--title"
        />
        <div>
          
        </div>
        
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`News Item Test get default NewsItem with data  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="news-item"
      >
        <div
          class="news-item__img"
        >
          <div
            class="news-item__img--content"
            style="background-image: url(https://www.irishtimes.com/polopoly_fs/1.4221277.1586082453!/image/image.jpg);"
            title="LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says"
          />
        </div>
        <div
          class="news-item__info"
        >
          <h3
            class="news-item__info--title"
          >
            LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says
          </h3>
          <div>
            <span
              class="news-item__info--source"
            >
              The Irish Times
            </span>
            <span
              class="news-item__info--published-at"
            >
              2 hours ago
            </span>
          </div>
          <span
            class="news-item__info--description"
          >
            70,000 applications made to Ireland on Call healthcare worker scheme
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="news-item"
    >
      <div
        class="news-item__img"
      >
        <div
          class="news-item__img--content"
          style="background-image: url(https://www.irishtimes.com/polopoly_fs/1.4221277.1586082453!/image/image.jpg);"
          title="LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says"
        />
      </div>
      <div
        class="news-item__info"
      >
        <h3
          class="news-item__info--title"
        >
          LIVE Coronavirus: 4,500 tests to be carried out per day from this week, HSE says
        </h3>
        <div>
          <span
            class="news-item__info--source"
          >
            The Irish Times
          </span>
          <span
            class="news-item__info--published-at"
          >
            2 hours ago
          </span>
        </div>
        <span
          class="news-item__info--description"
        >
          70,000 applications made to Ireland on Call healthcare worker scheme
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
